backend/code/archipelag/message/views.py

```diff
@@ -0,0 +1,25 @@
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from archipelag.message.forms import MessageForm
+from archipelag.market.models import Market
+#from archipelag.notification.tasks import send_notification_for_event
+
+
+@login_required
+def message_create(request, market_id):
```

Comment:
If you split this into many methods, for example:
- on_post
- on_valid
- on_get
The unit tests you would need to write will be simpler.